golang/json/handler.go

```diff
@@ -117,6 +117,33 @@ func Register(ch *tchannel.Channel, funcs Handlers, onError func(context.Context
 	return nil
 }
 
+func RegisterSub(ch *tchannel.SubChannel, funcs Handlers, onError func(context.Context, error)) error {
```

Comment:
Rather than do this, let's create a separate interface in tchannel that only contains "Register". That way, both tchannel.Channel and tchannel.Subchannel can be passed to the same Register function, rather than duplicating logic.

Maybe something like:
```go
type Registrar interface {
  func Register(h Handler, operationName string)
}
```